system/goos/dependencies_linux.go

```diff
@@ -1,6 +1,70 @@
 package goos
 
+import (
+	"log"
+	"os/exec"
+
+	"github.com/ironstar-io/tokaido/system/console"
+)
+
 // CheckDependencies - Root executable
 func CheckDependencies() *string {
+	errDocker := CheckDocker()
+	errDockerCompose := CheckDockerCompose()
+	errUnison := CheckUnison()
```

Comment:
IMO it should fail on the first instance of an error instead of processing all of them potentially arbitrarily then looking for error

Fail early is generally what I like to aim for, but this might be another golang best practice I’m not across

Reply:
I'd like to keep this method in this particular case because it tells the user of every component that is missing. Otherwise, they're likely to keep coming back to this and figure out they need one more component, you know? 

Otherwise I see this playing out: 

"OK, I need to go get Docker"... comes back 20 minutes later... "OK, now I need to get Docker Compose...".... "Argh! Now Unison?!"

At least this way they know about everything that's missing up-front.